nx/lib/nx/defn/expr.ex

```diff
@@ -764,13 +760,9 @@ defmodule Nx.Defn.Expr do
         description: """
         the do-block in while must return tensors with the same shape, type, and names as the initial arguments.
 
-        Body matches template:
-
-        #{inspect_as_template(body)}
-
-        and initial argument has template:
+        Comparison:
 
-        #{inspect_as_template(initial)}
+        #{Nx.Defn.TemplateDiff.build_and_inspect(body, initial)}
```

Comment:
Don’t we lose the context of which one is body and which one is initial? Can we say something like (green is body, red is intitial)?

Reply:
We can add some parameterized labelling.

Something like this:

`build_and_inspect(body, "body", initial, "initial")`

which would print:

```
<<<<< body <<<<<
green thing
==========
red thing
>>>>> initial >>>>>
```

or something similar

Reply:
That looks great to me, your call :)

Reply:
Nice, I'll apply this then. Makes it easier for non-colorized outputs too